OpenMTRDemo/MainForm.cs

```diff
@@ -7,38 +7,127 @@
 using System.Text;
 using System.Threading.Tasks;
 using System.Windows.Forms;
+using OpenCvSharp;
 using OpenMTR;
 
 namespace OpenMTRDemo
 {
     public partial class MainForm : Form
     {
-        OpenFileDialog FileBrowser;
-        Meter meter;
+        OpenFileDialog OpenBrowser;
+        SaveFileDialog SaveBrowser;
+        OpenMTR.Meter meter;
+
         public MainForm()
         {
-            FileBrowser = new OpenFileDialog();
-            FileBrowser.Filter = "Image Files|*.bmp;*.jpeg;*.jpg;*.gif";
+            OpenBrowser = new OpenFileDialog();
+            SaveBrowser = new SaveFileDialog();
+            OpenBrowser.Filter = "Image Files|*.bmp;*.gif;*.jpg;*.jpeg;*.png";
+            SaveBrowser.Filter = "JPG|*.jpg";
             InitializeComponent();
         }
 
-        private void loadToolStripMenuItem_Click(object sender, EventArgs e)
+        private void OpenToolStripMenuItem_Click(object sender, EventArgs e)
         {
-            if (FileBrowser.ShowDialog() == DialogResult.OK)
+            if (OpenBrowser.ShowDialog() == DialogResult.OK)
             {
-                meter = ReadData.GetMeter(FileBrowser.FileName);
+                meter = OMTR.GetMeter(OpenBrowser.FileName);
                 Render();
             }
+            MetaDataTextBox.Text = meter.MeterRead + "";
+            SetDisableableControls(true);
+            WidthTextBox.Text = meter.SourceImage.Width + " px";
+            HeightTextBox.Text = meter.SourceImage.Height + " px";
         }
 
         private void Render()
         {
-            inputImageBox.Image = DemoUtilities.MatToBitmap(meter.SourceImage);
+            InputImageBox.Image = DemoUtilities.MatToBitmap(meter.SourceImage);
             RenderOutput();
         }
         private void RenderOutput()
         {
-            outputImageBox.Image = DemoUtilities.MatToBitmap(meter.ModifiedImage);
+            meter.ModifiedImage = meter.SourceImage.Clone();
+            ApplyFilters(meter.ModifiedImage);
+            OutputImageBox.Image = DemoUtilities.MatToBitmap(meter.ModifiedImage);
+        }
+
+        private void ApplyFilters(Mat imageToFilter)
+        {
+            foreach (string filter in FilterListBox.SelectedItems)
+            {
+                switch (filter)
+                {
+                    case "Black and White":
+                        OMTR.GetBlackAndWhite(imageToFilter, imageToFilter);
```

Comment:
Just use ImageUtils.ColorToGray().

Reply:
Better Code Hub won't take it.

Reply:
if that's the case call the Cv2 static methods directly

Reply:
Can do.